snowflake/data_access_sync_from_target.go

```diff
@@ -17,17 +20,105 @@ import (
 	"github.com/raito-io/cli-plugin-snowflake/common"
 )
 
-func (s *AccessSyncer) importAllRolesOnAccountLevel(accessProviderHandler wrappers.AccessProviderHandler, repo dataAccessRepository, shares []string, configMap *config.ConfigMap) error {
-	externalGroupOwners := configMap.GetStringWithDefault(SfExternalIdentityStoreOwners, "")
-	excludedRoles := s.extractExcludeRoleList(configMap)
-	linkToExternalIdentityStoreGroups := configMap.GetBoolWithDefault(SfLinkToExternalIdentityStoreGroups, false)
+type AccessFromTargetSyncer struct {
+	configMap             *config.ConfigMap
+	repo                  dataAccessRepository
+	ctx                   context.Context
+	accessSyncer          *AccessSyncer
+	accessProviderHandler wrappers.AccessProviderHandler
+
+	shares                            []string
+	linkToExternalIdentityStoreGroups bool
+	externalGroupOwners               string
+	excludedRoles                     map[string]struct{}
+	lock                              sync.Mutex
+}
+
+func NewAccessFromTargetSyncer(ctx context.Context, accessSyncer *AccessSyncer, repo dataAccessRepository, accessProviderHandler wrappers.AccessProviderHandler, configMap *config.ConfigMap) *AccessFromTargetSyncer {
+	return &AccessFromTargetSyncer{
+		accessSyncer:          accessSyncer,
+		ctx:                   ctx,
+		configMap:             configMap,
+		repo:                  repo,
+		accessProviderHandler: accessProviderHandler,
+	}
+}
+
+func (s *AccessFromTargetSyncer) syncFromTarget() error {
+	s.externalGroupOwners = s.configMap.GetStringWithDefault(SfExternalIdentityStoreOwners, "")
+	s.excludedRoles = s.extractExcludeRoleList()
+	s.linkToExternalIdentityStoreGroups = s.configMap.GetBoolWithDefault(SfLinkToExternalIdentityStoreGroups, false)
+
+	defer func() {
+		logger.Info(fmt.Sprintf("Total snowflake query time:  %s", s.repo.TotalQueryTime()))
+		s.repo.Close()
```

Comment:
Small thing, but for me it feels more like we should close the repo in the place were we created the repo. (In this case in the `SyncAccessProvidersFromTarget` method of the `AccessSyncer` type.